frontend/src/components/pomodoroview/PomodoroView.jsx

```diff
@@ -69,7 +69,37 @@ function RemainingTimeContextProvider(props) {
   const [isClockPlaying, setIsClockPlaying] = useState(false);
   const [workDurationMinutes, setWorkDurationMinutes] = useState(25);
   const [breakDurationMinutes, setBreakDurationMinutes] = useState(5);
+  const [m, setM] = useState(25);
+  const [s, setS] = useState(0);
 
+  useEffect ( () => {
+    setData((data) => setStartTime(data), "startTime")
+    setData((data) => setIsUserBreaking(data == "true"), "isUserBreaking")
+    setData((data) => setClockKey(Number(data)), "ClockKey")
+    setData((data) => setIsClockPlaying(data == "true"), "isClockPlaying")
+    setData((data) => setWorkDurationMinutes(Number(data)), "workDurationMinutes")
+    setData((data) => setBreakDurationMinutes(Number(data)), "breakDurationMinutes")
+    setData((data) => setM(Number(data)), "M")
+    setData((data) => setS(Number(data)), "S")
+  }, [])
+
+  useEffect ( () => {
+    localStorage.setItem("startTime", startTime)
+    localStorage.setItem("isUserBreaking", isUserBreaking)
+    localStorage.setItem("ClockKey", ClockKey)
+    localStorage.setItem("isClockPlaying", isClockPlaying)
+    localStorage.setItem("workDurationMinutes", workDurationMinutes)
+    localStorage.setItem("breakDurationMinutes", breakDurationMinutes)
+    localStorage.setItem("M", m)
+    localStorage.setItem("S", s)
+  })
+
+  function setData(func, key) {
```

Comment:
setData sounds not so direct for me. and it could be more in consistent with localStorage.setItem by

```suggestion
  function localStorage_loadItem(itemName, itemSetter, itemType) {}
```

Reply:
I used the name loadItemFromLocalStorage.